src/base-client.ts

```diff
@@ -7,47 +7,41 @@ import { Logger } from './logger';
 import { Subscription, SubscriptionListeners, SubscriptionConstructor, replaceMissingListenersWithNoOps } from './subscription';
 import { BaseSubscription } from './base-subscription';
 import { createTokenProvidingStrategy } from './token-providing-subscription';
-import { createH2TransportStrategy } from './transports';
+import { createTransportStrategy } from './transports';
 import { ElementsHeaders, responseToHeadersObject } from './network';
 import { subscribeStrategyListenersFromSubscriptionListeners } from './subscribe-strategy';
+import WebSocketClient from './websocket-client';
 import * as PCancelable from 'p-cancelable';
 
+import WebSocketTransport from './transport/websocket';
+import HttpTransport from './transport/http';
+
 export interface BaseClientOptions {
     host: string;
     encrypted?: boolean;
     logger?: Logger;
 }
 
 export class BaseClient {
-    private baseURL: string;
+    private host: string;
     private XMLHttpRequest: any;
     private logger: Logger;
+    private websocketTransport: WebSocketTransport;
+
     constructor(private options: BaseClientOptions) {
-        let host = options.host.replace(/\/$/, '');
-        this.baseURL = `${options.encrypted !== false ? "https" : "http"}://${host}`;
+        this.host = options.host.replace(/\/$/, '');
         this.logger = options.logger;
-    }
 
-    private xhrConstructor: (path: string) => (headers: ElementsHeaders) => XMLHttpRequest = (path) => {
-        
-        return (headers) => {
-            const requestOptions: RequestOptions = {
-                method: "SUBSCRIBE",
-                path: path,
-                headers: headers
-            }
-    
-            return this.createXHR(this.baseURL, requestOptions);
-        }
+        this.websocketTransport = new WebSocketTransport(this.host);
     }
 
     public request(options: RequestOptions, tokenProvider?: TokenProvider, tokenParams?: any): PCancelable {
         if(tokenProvider){
-            return tokenProvider.fetchToken(tokenParams).then( token =>                
-                { 
-                    options.headers['Authorization'] = `Bearer ${token}`
+            return tokenProvider.fetchToken(tokenParams).then( token =>                  
+                {
+                    options.headers['Authorization'] = `Bearer: ${token}`
                     return executeNetworkRequest(
-                        () => this.createXHR(this.baseURL, options),
+                        () => new HttpTransport(this.host).request(options),
```

Comment:
It might be worth making the transport instance a member of the class instead of initiating it on every request.

Reply:
It's good point.. I gonna make it that way.